mpmp/utilities/tcga_utilities.py

```diff
@@ -375,3 +375,56 @@ def filter_to_cross_data_samples(X_df,
 
     return (X_filtered_df, y_filtered_df)
 
+
+def get_and_save_sample_info(tcga_df,
+                             sampletype_codes_dict,
+                             cancertype_codes_dict,
+                             training_data='expression'):
+    """Extract cancer type/sample type info from TCGA samples.
+
+    Also save info to a TSV file, to use for classification/analysis later.
+
+    Arguments
+    ---------
+    tcga_df (pd.DataFrame): df with sample IDs as index
+    sampletype_codes_dict (dict): maps last 2 digits of TCGA barcode to sample type
+    cancertype_codes_dict (dict): maps first 2 digits of TCGA barcode to cancer type
+    training_data (str): describes what type of data is being downloaded
+
+    Returns
+    -------
+    tcga_id (pd.DataFrame): df describing sample type/cancer type for included samples
+    """
+
+    # extract sample type in the order of the gene expression matrix
+    tcga_id = pd.DataFrame(tcga_df.index)
+
+    # extract the last two digits of the barcode and recode sample-type
+    tcga_id = tcga_id.assign(sample_type = tcga_id.sample_id.str[-2:])
+    tcga_id.sample_type = tcga_id.sample_type.replace(sampletype_codes_dict)
+
+    # extract the first two ID numbers after `TCGA-` and recode cancer-type
+    tcga_id = tcga_id.assign(cancer_type = tcga_id.sample_id.str[5:7])
```

Comment:
Is there a guarantee that these IDs will always work the same way, or should you split on TCGA and take the first two characters of the second string?

Reply:
Good idea. I doubt the TCGA ID format will change, but what you're suggesting seems safer/more robust.